### PR TITLE
Setting some pip versions

### PR DESCRIPTION
**Pull request by znado**
Modifying `setup.cfg` and `README.md` to set stricter pip package versions, and clarify some install instructions.

Updating everything to use a python version 3.9.x. If this is a problem for people, we can change it.

Removed the `ctcdecode` dep in `setup.cfg` because we need to install it separately, as described in [PR#70](https://github.com/mlcommons/algorithmic-efficiency/pull/70).

I confirmed the versions install successfully when running the following sequence of commands:
```
pip3 install -e '.[jax_cpu]'
pip3 install -e '.[full_jax]'
pip3 install -e '.[pytorch_cpu]'
pip install torch-scatter -f 'https://data.pyg.org/whl/torch-1.12.0+cu113.html'
pip3 install -e '.[full_pytorch]'
```

The final pip package list is:
```
~/algorithmic-efficiency$ pip list
Package                      Version      Editable project location
---------------------------- ------------ ----------------------------------
absl-py                      1.0.0
algorithmic-efficiency       0.0.1        /home/znado/algorithmic-efficiency
appdirs                      1.4.4
astunparse                   1.6.3
attrs                        22.1.0
audioread                    2.1.9
cached-property              1.5.2
cachetools                   5.2.0
certifi                      2022.6.15
cffi                         1.15.1
charset-normalizer           2.1.0
chex                         0.1.4
clu                          0.0.7
contextlib2                  21.6.0
cycler                       0.11.0
decorator                    5.1.1
dill                         0.3.5.1
dm-tree                      0.1.7
etils                        0.7.1
flatbuffers                  1.12
flax                         0.3.5
fonttools                    4.34.4
future                       0.18.2
gast                         0.4.0
google-auth                  2.10.0
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
googleapis-common-protos     1.56.4
grpcio                       1.47.0
h5py                         3.7.0
idna                         3.3
imageio                      2.21.1
importlib-metadata           4.12.0
importlib-resources          5.9.0
jarowinkler                  1.2.0
jax                          0.2.28
jaxlib                       0.1.76
joblib                       1.1.0
jraph                        0.0.2.dev0
keras                        2.9.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.4
Levenshtein                  0.20.0
libclang                     14.0.6
librosa                      0.9.2
llvmlite                     0.39.0
Markdown                     3.4.1
MarkupSafe                   2.1.1
matplotlib                   3.5.3
ml-collections               0.1.1
msgpack                      1.0.4
networkx                     2.8.5
numba                        0.56.0
numpy                        1.22.4
oauthlib                     3.2.0
opt-einsum                   3.3.0
optax                        0.0.9
packaging                    21.3
pandas                       1.4.3
Pillow                       9.2.0
pip                          22.0.2
pooch                        1.6.0
promise                      2.3
protobuf                     3.20.1
pyasn1                       0.4.8
pyasn1-modules               0.2.8
pycparser                    2.21
pyparsing                    3.0.9
python-dateutil              2.8.2
python-version               0.0.2
pytz                         2022.2
PyWavelets                   1.3.0
PyYAML                       6.0
rapidfuzz                    2.4.3
requests                     2.28.1
requests-oauthlib            1.3.1
resampy                      0.4.0
rsa                          4.9
scikit-image                 0.19.3
scikit-learn                 1.0.1
scipy                        1.9.0
sentencepiece                0.1.97
setuptools                   59.6.0
six                          1.16.0
SoundFile                    0.10.3.post1
tensorboard                  2.9.0
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.9.0
tensorflow-datasets          4.4.0
tensorflow-estimator         2.9.0
tensorflow-hub               0.12.0
tensorflow-io-gcs-filesystem 0.26.0
tensorflow-metadata          1.9.0
tensorflow-text              2.9.0
termcolor                    1.1.0
threadpoolctl                3.1.0
tifffile                     2022.8.8
toolz                        0.12.0
torch                        1.12.0
torch-scatter                2.0.9
torchvision                  0.13.0
tqdm                         4.64.0
typing_extensions            4.3.0
urllib3                      1.26.11
Werkzeug                     2.2.2
wheel                        0.37.1
wrapt                        1.14.1
zipp                         3.8.1
```

### CODE REVIEW

**Review by runame — CHANGES_REQUESTED** [review]
See my comments below.

I think it makes sense to just remove the `torch-scatter` requirement, since we only need a small piece of code from them and their source code is under the MIT license. I have created a PR #116 for this, which could be merged before merging this PR.

File: setup.cfg [hunk 1]
```diff
@@ -88,17 +99,18 @@ fastmri =
 ogbg =
   jraph==0.0.2.dev
   scikit-learn==1.0.1
-  clu==0.0.7
+  clu==0.0.7  # TODO(znado): Delete?
```
[the remark this answers is not in the document]
Comment (znado) [reply setup.cfg:102]: Ah right, deleted the comment.

File: setup.cfg [hunk 2]
```diff
@@ -57,29 +57,40 @@ python_requires = >=3.7
...
-full_dev =
-  %(full)s
+full_dev_jax =
+  %(full_jax)s
   %(dev)s
 
+full_pytorch =
+  %(criteo1tb)s
+  %(fastmri)s
+  %(ogbg_pytorch)s
```
[the remark this answers is not in the document]
Comment (znado) [reply setup.cfg:74]: originally I wanted to avoid needing torch-scatter when just using jax, but now that it is gone, we can just keep them the same.

**Comment by znado** [comment]
> See my comments below.
> 
> I think it makes sense to just remove the `torch-scatter` requirement, since we only need a small piece of code from them and their source code is under the MIT license. I have created a PR #116 for this, which could be merged before merging this PR.

Merged that PR, and undid separating setup.cfg into _jax and _pytorch.

**Comment by runame** [comment]
Great, thanks! This is really useful, the setup instructions were confusing before. Newcomers should have a much better time now.

One last thing: the instructions in the README have to be adjusted to the undoing of the jax/pytorch requirements separation in `setup.cfg`.

**Comment by znado** [comment]
ah good catch, done

**Review by runame — APPROVED** [review]

**Pull request merged**